### ISSUE TITLE
[Bug] My VA - Onsite Notifications - iOS - Styling anomaly

### ISSUE THREAD

**Issue by tlei123** [department-of-veterans-affairs/va.gov-team#45144]
## What happened?
On **Staging My VA** page, in **Notifications** section, the **new-debt** alert's **Close** button collided with **Manage your VA debt** link.

**NOTE**: Issue replication & fix in Staging _may_ need test-users' Staging data-states restored after QA test-runs are completed. [See [Test-cases & accounts doc](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/staging-test-accounts-onsite-notification.md#test-users-for-end-to-end-testing-with-dmc) &mdash; use a test-user that doesn't have a burnt-date under Burnt column]

## Specs:
- Device: Apple iPhone w/ viewport-width  < 402px\* [any model after 2018 iPhone SE]
- Browser: Safari & Chrome [both on iOS] &mdash; Chrome on Android is fine even at 360px viewport-width.
- Test User _(if applicable)_: [See [Test-cases & users doc](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/staging-test-accounts-onsite-notification.md#test-users-for-end-to-end-testing-with-dmc).]

\*In Chrome devtools responsive emulation, issue's replicable between 382px and 402px (inclusive).

## Steps to Reproduce
- URL: https://staging.va.gov/my-va
- Test User _(if applicable)_: Any [test-user](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/staging-test-accounts-onsite-notification.md#test-users-for-end-to-end-testing-with-dmc) that isn't burnt

### Screenshot(s)

- Mobile:
  - (1) Alert Close button collides with Manage link: 
    ![image](https://user-images.githubusercontent.com/587583/182262852-5c59a838-6e35-4c9f-b783-7046f6219539.png)

## Desired content/styling
- [ ] (1) **New-debt** alert Close button should not collide with any alert-content.

## Acceptance Criteria
[See **Desired content/styling** above.]

## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to Epic** (what body of work is this a part of? possibly `Ongoing Maintenance`)
- [x] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `triage`, `tools-improvements`)
- [x] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [x] **Labeled with `Bug`**

**Comment by tmitchellgcio** [comment]
@tlei123 what are you using to test these? An actual device or something else? I use the chrome emulator and I get the view expected. I did notice the pixel 5 might need some readjustment. Will have to research as I haven't had to code for specific phones before just specific viewports. I may need to readjust the viewport to change after 410px. Regardless let me know what your using for emulation so I can test it the exact same way.
![iphone 5se.png](https://images.zenhubusercontent.com/60ad368786f922a382c8b16b/64c8df57-00b9-42d6-b378-aaa14c2f2ed4)
![iphone 6.png](https://images.zenhubusercontent.com/60ad368786f922a382c8b16b/1d7072a7-674c-4f95-86b3-1f326c6ad733)
![iphone se.png](https://images.zenhubusercontent.com/60ad368786f922a382c8b16b/8fae0978-2123-43d9-8622-c8a487bad5f9)

![pixel5.png](https://images.zenhubusercontent.com/60ad368786f922a382c8b16b/3854851b-7f77-477e-acc6-7b18ccd3682a)

**Comment by tlei123** [comment]
@tmitchellgcio , i'm Closing this as it's not reliably replicable.

**TLDR**: Actual cross-device testing's still preferable, but I only have an iOS iPhone 12 mini.

i just did some double-checking and something about BrowserStack ain't making sense.  BrowserStack shows this bug on a _wider-viewport_ iPhone 12, but not on the narrower iPhone 12 mini.  :::scratching head:::

**BrowserStack iPhone 12 Safari** [390px viewport-width]:
![image](https://user-images.githubusercontent.com/587583/182944687-fd6ff767-ece9-4b8b-8145-1e8888954f51.png)

I have an iPhone 12 mini, and BrowserStack's mini isn't showing this issue even though viewport-width is _narrower_!

**BrowserStack iPhone 12 mini Safari** [375px viewport-width]:
![image](https://user-images.githubusercontent.com/587583/182947672-d6d26f0a-ff3a-4d56-a5c8-9850e80fa25a.png)

**My own iPhone 12 mini Safari**:
![image](https://user-images.githubusercontent.com/587583/182948337-f34ef2de-52ef-49e5-8c44-af77e946e0a6.png)

@pjhill @ddzz , do you know if anyone has a real iPhone 12 [not mini]?

**Comment by tlei123** [comment]
Sorry, @tmitchellgcio !  Just re-opened this.  Above comment updated, and confirming here that this bug's replicable in Chrome responsive emulation at 390px width:
![image](https://user-images.githubusercontent.com/587583/182959412-0408a40d-8137-4034-a8c6-d3295cecb7d3.png)

@pjhill @ddzz , NVM about finding someone with a real iPhone 12.  Sorry for the confusion!

[Note to self: Don't QA while suffering a pounding headache!]

**Comment by tlei123** [comment]
^ @tmitchellgcio , this looks like Design-Systems stuff, doesn't it?

i haven't popped the hood on their code, and prob'ly should wait to do that tomorrow 'cuz my brain's still kinda mushy right now.

**Comment by tmitchellgcio** [comment]
I am not 100% but I think this is something I can override. I will look into adjusting this today as I believe it's something I can fix. I noticed the issue on the pixel on chrome and then responsive so yeah around 410 or so I am thinking I will try to set the media filter to swap it to a new line. If it's something I can't fix I will let you know but hopefully I can.

**Comment by k80bowman** [comment]
Hello! Just popping in here from the Design System team. We recently merged a fix for the icon overlapping issue, I'll be bumping the version of the component library in vets-website here shortly so you can get that fix.

I wonder, however, if the problem here could also be fixed by making the "You have new debt" text into a heading (looks like an `h3` might be appropriate here) instead of a span. That way the "Manage your VA debt" text would always appear on the next line instead of relying on wrapping. In our [guidance on Alerts](https://design.va.gov/components/alert#heading-level) we do say that they should always contain a heading.

**Comment by tlei123** [comment]
Thanks, @k80bowman !  @cgednalske @tmitchellgcio , just let me know if you do make the Manage link it's own line/heading.  Officially, My QA half's solution-agnostic, although unofficially my Dev half prefers the link on it's own line since it's likely more future-proof.

**Comment by ajakabcin** [comment]
thanks @k80bowman - what is "shortly" here? We're thinking we should see what your update looks like and then if we still need to make a change we'll go from there. 

cc @tmitchellgcio

**Comment by k80bowman** [comment]
@ajakabcin I was hoping to get the change in on Friday morning, but it seems a couple changes we made to our components have caused Cypress tests to fail, so I am fixing those. Hopefully will get them resolved today.

**Comment by ajakabcin** [comment]
thanks @k80bowman! I think we'll wait for the design system update first and then see if we need to make any other changes to address this issue.

@ericasauv09 let's hold off on pulling this into sprint 80 then until we see what comes through from the design system.

**Comment by ajakabcin** [comment]
Moved back to backlog - will wait for updates from design

**Comment by bkjohnson** [comment]
The version of the component-library with the alert icon fixes is [now available in `vets-website`](https://github.com/department-of-veterans-affairs/vets-website/pull/21865), so you should be unblocked.

**Comment by ericasauv09** [comment]
@ajakabcin should we pull this back into the sprint?

**Comment by tmitchellgcio** [comment]
We should validate it. I would assume we shouldn't have to make any changes to the component ourselves right @bkjohnson

**Comment by bkjohnson** [comment]
That is correct.

**Comment by ajakabcin** [comment]
Yes @tmitchellgcio let's see how it looks now - can you and @aagosto90 take a look and let us know if it needs additional work on our side?

@ericasauv09 if it needs more work then yes let's pull it in :)

**Comment by aagosto90** [comment]
@ajakabcin @tmitchellgcio @ericasauv09 Confirmed that in Chrome at 390px wide like Tze was looking at it, this bug is fixed
![Screen Shot 2022-08-11 at 12 41 38 PM](https://user-images.githubusercontent.com/97965610/184187938-273d3fbf-3995-49d3-a2ec-305ec8f7c28f.png)

**Comment by ajakabcin** [comment]
Thank you @aagosto90! 

@tlei123 I'm going to close this now - thanks for flagging!
